src/videojs-contrib-hls.js

```diff
@@ -52,8 +53,21 @@ const Hls = {
   xhr: xhrFactory()
 };
 
-// the desired length of video to maintain in the buffer, in seconds
-Hls.GOAL_BUFFER_LENGTH = 30;
+Object.defineProperty(Hls, 'GOAL_BUFFER_LENGTH', {
+  get() {
+    videojs.log.warn('Hls.GOAL_BUFFER_LENGTH is deprecated and should not be used');
```

Comment:
I don't think it should be deprecated if we don't provide any other way of setting. It's UNSAFE but not necessarily deprecated!